components/places/src/bookmark_sync/store.rs

```diff
@@ -822,8 +827,9 @@ impl<'a> dogear::Store<Error> for Merger<'a> {
         let descendants = root.descendants();
         let deletions = deletions.collect::<Vec<_>>();
 
-        let tx = self.store.db.unchecked_transaction()?;
-        self.store.update_local_items(descendants, deletions)?;
+        let mut tx = self.store.db.time_chunked_transaction()?;
+        self.store
+            .update_local_items(descendants, deletions, &mut tx)?;
         self.store.stage_local_items_to_upload()?;
```

Comment:
(The reason we can't `maybe_commit` between `update_local_items` and `stage_local_items_to_upload` is because `stage_local_items_to_upload` stages outgoing bookmarks using the change counter. At the end of a merge, everything that needs to be uploaded has `changeCounter = 1`. If the main connection sneaks in a write between the time we call `update_local_items`, and `stage_local_items_to_upload`, it might bump the change counter for items that we haven't merged yet, and upload inconsistent state to the server).